archinstall/lib/hardware.py

```diff
@@ -210,6 +210,18 @@ def graphics_devices(self) -> dict[str, str]:
 
 
 class SysInfo:
+	@staticmethod
+	def has_battery() -> bool:
```

Comment:
For a future PR, we can move this into _SysInfo and cache the return value so it doesn't get calculated multiple times